### ISSUE TITLE
Functionality that is useful for Gitea's administrative workflow

### ISSUE THREAD

**Issue by guillep2k** [go-gitea/gitea#8659]
As the move to gitea.com is getting closer, I was thinking about those useful features that we as maintainers will miss after we move; I mean **those specifically needed for the workflow of maintaining Gitea itself**.

**The idea is to make a list of those features we consider the most useful, so contributors can have specific goals to help with.**

As a maintainer I'm not familiar with many of the tasks involved in the administration of Gitea, so I'll start with my short list to open the discussion.

### NOTE: in this list I'm collecting ideas from the comments below

- [x] An `Update branch` button #9750 (#9784).
- [x] Expand/collapse source files while reviewing code #8579 (#8924).
- [x] The ability to expand code beyond the given context when comparing/reviewing (#8924).
- [x] Mark files as seen while reviewing code. (#11730) -> (#19007)
- [x] Close/reopen issues from PR comments (#8866).
- [x] Make changelog tool use Gitea ([gitea/changelog#19](https://gitea.com/gitea/changelog/pulls/19)).
- [x] Allow edits from maintainer (#18002)
- [x] Require branches to be up to date before merging #9786 (#11012).
- [ ] Show on Pull request list page:
    - [x] Number of required approvals still remaining, or the number approvals obtained (#8762) (#9274) (#10238).
    - [x] Has the PR passed CI.
    - [x] Does it contain conflicts (#10130).
    - [ ] Is it blocked by another issue/PR. (#29117)
- [ ] Commiteable code change suggestions (#9388)
- [ ] Multi line review comments (#12640, #25523)
- [x] Select specific commits when reviewing code.  #25528
- [x] Always display the members page for public organizations; see #7501

**Comment by davidsvantesson** [comment]
> * [ ]  A count of approvals in the PR page.

You already have that?
 _This Pull Request doesn't have enough approvals yet. 1 of 2 approvals granted._
Or do you mean the list of reviewers in the right sidebar? That I think would be useful to have.

**Comment by zeripath** [comment]
He means on the list of PRs page. In GitHub we use labels to set this but Gitea has that built in

**Comment by lunny** [comment]
It could be `Show approvals counts on Pull request list page"

**Comment by guillep2k** [comment]
#### Add more information on PR list to streamline administration:

* Number of required approvals still remaining.
* Is it blocked by another issue/PR.
* Has passed CI.
* (Much harder, I think) Does it produce conflicts.

**Comment by guillep2k** [comment]
For this one:

> - [ ] Does it produce conflicts.

I think a small client-side button to check could be used? I mean, it could show something like ❔ that, when pressed, a request is sent to the server and the button is changed into ✔️ or ❌.

**Comment by blueworrybear** [comment]
It's better to add `Expand Up` and `Expand Down` button too IMO. 🤔

**Comment by guillep2k** [comment]
> 
> 
> It's better to add `Expand Up` and `Expand Down` button too IMO. 🤔

Sorry, are you referring to the "does it produce conflicts" option?

**Comment by jamesorlakin** [comment]
> Number of required approvals still remaining, or the number approvals obtained.

I came across this issue seeing if this exact feature was possibly in progress already, or whether it was worth me having an attempt! I've never touched Go before (I'm a full-stack TypeScript dev) but love the work put into Gitea.

After some playing about for over an hour I have this:
![image](https://user-images.githubusercontent.com/7294642/67897634-bb524100-fb56-11e9-875b-1d470c5c3124.png)

The first number being the number of granted approvals, and the second the required ones. Let me know if you would like a PR 👍 .

**Comment by lafriks** [comment]
@jamesorlakin sure send PR ;)

**Comment by jamesorlakin** [comment]
@lafriks Sure thing, I'll try to get it out tomorrow

**Comment by blueworrybear** [comment]
@guillep2k 
I mean the code folding/unfolding. Here is an example:
![expand](https://i.imgur.com/uTR2qkY.png)

**Comment by guillep2k** [comment]
> 
> 
> @guillep2k
> I mean the code folding/unfolding. Here is an example:

Oh, of course!

**Comment by jolheiser** [comment]
Changelog tool now supports Gitea. 🎉

**Comment by jamesorlakin** [comment]
> - [ ] Does it produce conflicts. _(Much harder, probably)_

Would this just mean making use of the ConflictedFiles array on the pull model? If so, I whipped up this in a few lines by tweaking the template:
![image](https://user-images.githubusercontent.com/7294642/73613483-05cbd980-45ee-11ea-8c87-57c8dbe82a9a.png)

**Comment by guillep2k** [comment]
> 
> 
> Would this just mean making use of the ConflictedFiles array on the pull model? If so, I whipped up this in a few lines by tweaking the template:

@jamesorlakin Care to submit a PR? 😁

**Comment by lunny** [comment]
It seems it's easier to diaplsy files conflicted.

**Comment by zeripath** [comment]
This is on the PRs list page not on a particular pr page.

**Comment by 6543** [comment]
@guillep2k you can add #11012

**Comment by 6543** [comment]
@guillep2k you can check `Require branches to be up to date before merging` by the way :)

**Comment by 6543** [comment]
wow now it's near of complete :tada:

**Comment by 6543** [comment]
> Select specific commits when reviewing code.

can this be specified bit more please?

**Comment by delvh** [comment]
Also, what I noticed is missing:
- [ ] multiline comments
- [ ] comments on commits as well as in PRs
